parser/lexer.go

```diff
@@ -165,7 +165,8 @@ type token struct {
 	loc ast.LocationRange
 }
 
-type tokens []token
+// Tokens is a slice of token structs.
+type Tokens []token
```

Comment:
Any reason for this to be exported?

Reply:
```bash
$ golint -min_confidence 0
⋮
lexer.go:685:36: exported func Lex returns unexported type parser.tokens, which can be annoying to use
```

Reply:
I left the warning in place for a couple of other cases. Happy to revert if you prefer.

Reply:
It's ok.

Reply:
What are the implications of `Tokens` being public but `token` being private?  Does it mean all you can do with Tokens is to give it to the parser?

Reply:
Yeah, I think so. Tokens is probably completely opaque outside this package.